src/components/reusable/progressBar/progressBar.scss

```diff
@@ -0,0 +1,168 @@
+$progress-bar-background: var(--kd-color-background-inverse-hover);
+$progress-bar-indeterminate-color: map-get($progress-bar-status-colors, default);
+
+@mixin progress-bar-color($color) {
+    background-color: $color;
+}
+
+:host {
+    display: inline-block;
+    width: 100%;
+}
+
+::slotted([slot="unnamed"]) {
+    display: inline-block;
+    line-height: 8px;
+    margin-left: 0;
+    margin-top: 1px;
+}
+
+.progress-bar {
+    &__upper-container {
+        display: flex;
+        justify-content: space-between;
+        align-items: center;
+        padding: 0;
+        margin: 0;
+        height: 18px;
+    }
+
+    &__label {
+        display: flex;
+        align-items: center;
+        margin: 0;
+    }
+
+    &__helper-text {
+        margin: 0;
+        line-height: 16px;
+        height: 16px;
+    }
+
+    &__status-icon {
+        display: flex;
+        align-items: center;
+        padding-right: 2px;
+        line-height: 1;
+
+        p {
+            color: var(--kd-color-text-secondary);
+            padding: 0;
+            margin: 0;
+            font-size: 12px;
+            display: flex;
+            align-items: center;
+
+            span {
+                margin: 0 8px;
+                font-weight: 400;
+                vertical-align: middle;
+            }
+        }
+
+        kd-icon {
+            animation: fadeInRotate 600ms ease-in-out forwards;
+        }
+    }
+
+    &__active {
+        animation: active-slide 400ms linear infinite;
+    }
+
+    &__container {
+        width: 100%;
+        margin: 8px 0;
+        height: $progress-bar-height;
+        background-color: $progress-bar-background;
+        border-radius: $progress-bar-radius;
+        position: relative;
+        overflow: hidden;
+    }
+
+    &__background {
+        width: 100%;
+        height: 100%;
+        background-color: $progress-bar-background;
+        border-radius: $progress-bar-radius;
+        overflow: hidden;
+        position: relative;
+    }
+
+    &__main {
+        height: 100%;
+        background-color: map-get($progress-bar-status-colors, default);
+        border-radius: 0 $progress-bar-radius $progress-bar-radius 0;
+        transition: width 0.3s ease-in-out;
+
+        &.is-indeterminate {
+            width: 55px;
+            position: absolute;
+            left: 0;
+            @include progress-bar-color(map-get($progress-bar-status-colors, default));
+            animation: indeterminate 3s linear infinite;
+        }
+
+        @include progress-bar-color(map-get($progress-bar-status-colors, default));
+    }
+
+    &__success {
+        @include progress-bar-color(map-get($progress-bar-status-colors, success));
+    }
+
+    &__error {
+        @include progress-bar-color(map-get($progress-bar-status-colors, error));
+    }
+
+    &__active {
+        border-top-right-radius: 0;
+        border-bottom-right-radius: 0;
+    }
+}
+
+@keyframes indeterminate {
+    0% {
+        left: 0;
```

Comment:
This animation is fine as is IMO, but animating a property like left is probably not super smooth compared to transform for example especially on low-end devices.

Reply:
lastly, `translateX()` is a bit tricky here, as it executes the animation based on the element's width that it's attached to, not the parent's width.

it would take some restructuring, but i think i could make it work. i understand that it leads to better performance.

Reply:
Right, I had the same thought, just wanted to note that. Don't waste any time on it if it doesn't make sense. It does work fine as is.

Reply:
All good. These are all good things to consider, thank you